### PR TITLE
Refactor api

### PR DESCRIPTION
**Pull request by dbrinegar**
this is a new bootstrap-monitors.json for the api refactor

part of lightbend/es-backend#278

### CODE REVIEW

File: enterprise-suite/es-monitor-api/bootstrap-monitors.json [hunk 1]
```diff
@@ -1,639 +1,738 @@
...
+          "metric": "kube_pod_container_restarts_rate",
+          "window": "5m",
+          "confidence": "1",
+          "severity": {
+            "warning": {
+              "comparator": ">",
+              "threshold": "0"
+            }
+          },
+          "summary": "container restarting rapidly",
```
Comment (srikanthops) [inline enterprise-suite/es-monitor-api/bootstrap-monitors.json]: what happens if a container restarts only once..

File: enterprise-suite/es-monitor-api/bootstrap-monitors.json [hunk 2]
```diff
@@ -1,639 +1,738 @@
...
+          "description": "container {{$labels.container}} in pod {{$labels.pod}} of {{$labels.es_workload}} restarting rapidly"
+        }
+      },
+      "kube_pod_not_ready": {
+        "monitorVersion": "1",
+        "model": "threshold",
+        "parameters": {
+          "metric": "kube_pod_ready",
+          "window": "5m",
+          "confidence": "1",
```
Comment (srikanthops) [inline enterprise-suite/es-monitor-api/bootstrap-monitors.json]: confidence changed from 0.5 to 1

File: enterprise-suite/es-monitor-api/bootstrap-monitors.json [hunk 3]
```diff
@@ -1,639 +1,738 @@
...
+          "period": "5m",
+          "minslope": "1",
+          "confidence": "1",
+          "severity": {
+            "warning": {
+              "window": "5m"
+            }
+          },
+          "summary": "consumergroup falling behind",
+          "description": "{{$labels.consumergroup}} is falling behind"
```
Comment (srikanthops) [inline enterprise-suite/es-monitor-api/bootstrap-monitors.json]: QQ why don't have we have es_workload for all descriptions ?

File: enterprise-suite/es-monitor-api/bootstrap-monitors.json (hunk 2 above)
Comment (dbrinegar) [reply enterprise-suite/es-monitor-api/bootstrap-monitors.json]: look through the "dial fail monitors to 100% confidence" change in the commit log to understand that one.

File: enterprise-suite/es-monitor-api/bootstrap-monitors.json (hunk 1 above)
Comment (dbrinegar) [reply enterprise-suite/es-monitor-api/bootstrap-monitors.json]: nothing.  This requires all the data in a 5 minute window to have rate[1m] > 0

File: enterprise-suite/es-monitor-api/bootstrap-monitors.json (hunk 3 above)
Comment (dbrinegar) [reply enterprise-suite/es-monitor-api/bootstrap-monitors.json]: no descriptions or summaries were changed in this PR, this is just a JSON format change

Here the workload would be "consumer_group_exporter"

File: enterprise-suite/es-monitor-api/bootstrap-monitors.json [hunk 4]
```diff
@@ -1,639 +1,738 @@
...
+          "window": "5m",
+          "confidence": "1",
+          "severity": {
+            "critical": {
+              "comparator": "!=",
+              "threshold": "1"
+            }
+          },
+          "summary": "prometheus bad config",
+          "description": "current config for prometheus has errors, will prevent restarts"
```
[the remark this answers is not in the document]
Comment (dbrinegar) [reply enterprise-suite/es-monitor-api/bootstrap-monitors.json]: You... click the red pod in the UI?

Again, none of this was changed in this PR so I would prefer not to open scope to making alertmanager workable. Let's keep that separate.

File: enterprise-suite/es-monitor-api/bootstrap-monitors.json [hunk 5]
```diff
@@ -1,639 +1,738 @@
...
+      },
+      "zookeeper_open_file_descriptor": {
+        "monitorVersion": "1",
+        "model": "sma",
+        "parameters": {
+          "metric": "zk_open_file_descriptor_count",
+          "period": "15m",
+          "minval": "10",
+          "window": "15m",
+          "confidence": "1",
```
[the remark this answers is not in the document]
Comment (dbrinegar) [reply enterprise-suite/es-monitor-api/bootstrap-monitors.json]: that's how it is in master, was not changed in this PR

**Comment by dbrinegar** [comment]
@srikanthops the only parameter changes are here

https://github.com/lightbend/helm-charts/pull/109/commits/c899cb1271be8edb0807aa72df4c3a0317219cc6

the rest of this PR is json format

**Comment by dbrinegar** [comment]
this goes in hand with new es-monitor-api and the corresponding UI client, waiting for those before merge

**Comment by jsravn** [comment]
@dbrinegar FYI the monitor file was renamed to 'default-monitors.json'.

File: enterprise-suite/tests/smoke_prometheus [hunk 6]
```diff
@@ -237,17 +248,40 @@ busy_wait app_data
 
 prom_has_data "$app_instances"
 
+# make_monitor "type/name" "metric"
+make_monitor() {
+  curl -XPOST -H "Content-Type: application/json" -H 'Author-Name: Me' -H 'Author-Email: me@lightbend.com' -H 'Message: testing!' \
+    "$ES_MONITOR_API/monitors/$1" \
+    --data @- <<EOF
+    {
```
Comment (marcoderama) [inline enterprise-suite/tests/smoke_prometheus]: Might want to put that in a file in the `resources` dir.  Or not.

Comment (dbrinegar) [reply enterprise-suite/tests/smoke_prometheus]: oh! good idea

**Pull request merged**